app/post_test.go

```diff
@@ -297,6 +297,34 @@ func TestImageProxy(t *testing.T) {
 	}
 }
 
+func BenchmarkForceHTMLEncodingToUTF8(b *testing.B) {
+	HTML := `
+		<html>
+			<head>
+				<meta property="og:url" content="https://example.com/apps/mattermost">
+				<meta property="og:image" content="https://images.example.com/image.png">
+			</head>
+		</html>
+	`
+	ContentType := "text/html; utf-8"
```

Comment:
Is it possible to check to see if the content is already utf-8? It seems silly to pay a performance cost when most content is already utf-8 encoded.

Reply:
I would say checking the content type and not re-parsing if it's utf-8 is probably a good idea.

Reply:
what if the page does not provide content-type from header, but from its content. or from both side
`<meta charset="windows-1250"/>` or `<meta http-equiv="content-type" content="text/html; windows-1250">`

golang already have an implementation for these cases and I used it.

Reply:
@crspeller you mean add more check if there is UTF-8 identifier exist in the header?

Seems `DetermineEncoding()` is already doing it

Reply:
I don't really care how we detect it. I just noticed that in your test here the encoding is already utf-8. Since that's the most common case, it would be best if we didn't have the performance hit then.

Reply:
I added a few lines in the last commit to check if content-type header has `charset=utf-8`

But this only prevents reading first 1024 bytes compared to the previous one.

Reply:
Well, the result is not changed.
```
Running tool: /home/comet/go-v1.9.2/bin/go test -benchmem -run=^$ github.com/mattermost/mattermost-server/app -bench ^BenchmarkForceHTMLEncodingToUTF8$

[02:10:01 KST 2018/04/24] [INFO] (github.com/mattermost/mattermost-server/app.TestMain:28) -test.run used, not creating temporary containers
goos: linux
goarch: amd64
pkg: github.com/mattermost/mattermost-server/app
BenchmarkForceHTMLEncodingToUTF8/with_converting-4         	  100000	     11443 ns/op	   18752 B/op	      33 allocs/op
BenchmarkForceHTMLEncodingToUTF8/without_converting-4      	  300000	      3940 ns/op	    4632 B/op	      13 allocs/op
PASS
ok  	github.com/mattermost/mattermost-server/app	2.748s
Success: Benchmarks passed.
```

Reply:
So I guess the internal implementation checks and that is causing the performance difference. Alright. Let's just revert to the original implementation and that should be fine.

Reply:
@crspeller 👌